src/coreclr/vm/codeman.cpp

```diff
@@ -1338,531 +1277,262 @@ void EEJitManager::SetCpuInfo()
+        CPUCompileFlags.Set(InstructionSet_PCLMULQDQ);
     }
 
-    if (!CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableLZCNT))
+    if (((cpuFeatures & XArchIntrinsicConstants_Movbe) != 0) && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableMOVBE))
     {
-        CPUCompileFlags.Clear(InstructionSet_LZCNT);
+        CPUCompileFlags.Set(InstructionSet_MOVBE);
     }
 
-    if (!CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnablePCLMULQDQ))
+    if (((cpuFeatures & XArchIntrinsicConstants_Popcnt) != 0) && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnablePOPCNT))
     {
-        CPUCompileFlags.Clear(InstructionSet_PCLMULQDQ);
+        CPUCompileFlags.Set(InstructionSet_POPCNT);
     }
 
-    if (!CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableMOVBE))
+    // We need to additionally check that EXTERNAL_EnableSSE3_4 is set, as that
+    // is a prexisting config flag that controls the SSE3+ ISAs
+    if (((cpuFeatures & XArchIntrinsicConstants_Sse3) != 0) &&
+        CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableSSE3) &&
+        CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableSSE3_4))
     {
-        CPUCompileFlags.Clear(InstructionSet_MOVBE);
+        CPUCompileFlags.Set(InstructionSet_SSE3);
     }
 
-    if (!CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnablePOPCNT))
+    if (((cpuFeatures & XArchIntrinsicConstants_Sse41) != 0) && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableSSE41))
     {
-        CPUCompileFlags.Clear(InstructionSet_POPCNT);
+        CPUCompileFlags.Set(InstructionSet_SSE41);
     }
 
-    if (!CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableSSE))
+    if (((cpuFeatures & XArchIntrinsicConstants_Sse42) != 0) && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableSSE42))
     {
-        CPUCompileFlags.Clear(InstructionSet_SSE);
+        CPUCompileFlags.Set(InstructionSet_SSE42);
     }
 
-    if (!CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableSSE2))
+    if (((cpuFeatures & XArchIntrinsicConstants_Ssse3) != 0) && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableSSSE3))
     {
-        CPUCompileFlags.Clear(InstructionSet_SSE2);
+        CPUCompileFlags.Set(InstructionSet_SSSE3);
     }
 
-    // We need to additionally check that EXTERNAL_EnableSSE3_4 is set, as that
-    // is a prexisting config flag that controls the SSE3+ ISAs
-    if (!CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableSSE3) ||
-        !CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableSSE3_4))
+    if (((cpuFeatures & XArchIntrinsicConstants_Serialize) != 0) && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableX86Serialize))
     {
-        CPUCompileFlags.Clear(InstructionSet_SSE3);
+        CPUCompileFlags.Set(InstructionSet_X86Serialize);
     }
+#elif defined(TARGET_ARM64)
 
-    if (!CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableSSE41))
+#if !defined(TARGET_WINDOWS)
+    // Linux may still support no AdvSimd
+    if ((cpuFeatures & InstructionSet_AdvSimd) == 0)
```

Comment:
Should this just check for `InstructionSet_VectorT128` like the xarch path does?

Reply:
Makes sense.